driver/src/bin/main.rs

```diff
@@ -27,7 +27,7 @@ fn main() {
     };
     loop {
         // Start driver_components
-        run_driver_components(&db_instance, &contract, &mut price_finder);
+        run_driver_components(&db_instance, &contract, &mut *price_finder);
```

Comment:
It's a bit ugly to reference and dereference in the same line, but I also don't know a better way of doing this.

Reply:
This was what the warning suggested as a solution. I also don't know another way.